app_flutter/lib/agent_health_details.dart

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import 'package:cocoon_service/protos.dart' show Agent;
+
+/// A helper class for [Agent.healthDetails] that splits the expected summary
+/// with regex to get actionable health metrics.
+///
+/// Expected response:
+/// ```
+/// ssh-connectivity: succeeded
+///     Last known IP address: 192.168.1.29
+///
+/// android-device-ZY223D6B7B: succeeded
+/// has-healthy-devices: succeeded
+///     Found 1 healthy devices
+///
+/// cocoon-authentication: succeeded
+/// cocoon-connection: succeeded
+/// able-to-perform-health-check: succeeded
+/// ```
+class AgentHealthDetails {
+  factory AgentHealthDetails(Agent agent) {
+    final DateTime currentTime = DateTime.now();
+    final DateTime agentTime =
+        DateTime.fromMillisecondsSinceEpoch(agent.healthCheckTimestamp.toInt());
+    final Duration agentLastUpdateDuration = currentTime.difference(agentTime);
+
+    final String healthDetails = agent.healthDetails;
+    return AgentHealthDetails._(
+      agent,
+      healthDetails.contains(_hasHealthyDevices),
+      healthDetails.contains(_cocoonAuthentication),
+      healthDetails.contains(_cocoonConnection),
+      healthDetails.contains(_ableToPerformhealthCheck),
+      healthDetails.contains(_hasSshConnectivity),
+      agentLastUpdateDuration.inMinutes < minutesUntilAgentIsUnresponsive,
+    );
+  }
+
+  AgentHealthDetails._(
+    this.agent,
+    this.hasHealthyDevices,
+    this.cocoonAuthentication,
+    this.cocoonConnection,
+    this.canPerformHealthCheck,
+    this.hasSshConnectivity,
+    this.pingedRecently,
+  ) {
+    isHealthy = agent.isHealthy &&
+        hasHealthyDevices &&
+        cocoonAuthentication &&
+        cocoonAuthentication &&
+        canPerformHealthCheck &&
+        hasSshConnectivity &&
+        pingedRecently;
+  }
+
+  @visibleForTesting
+  static const int minutesUntilAgentIsUnresponsive = 10;
+
+  static final RegExp _hasSshConnectivity =
+      RegExp('ssh-connectivity: succeeded');
```

Comment:
Do these need to be regexps, or could they just be Strings?

Reply:
Good catch! Updated.